cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -747,15 +747,13 @@ class TableSorter {
     auto& comparator = comparator_;
     const auto& first_sort_key = sort_keys_[0];
 
-    ChunkLocation left_loc{0, 0};
-    ChunkLocation right_loc{0, 0};
+    ChunkLocation left_loc;
```

Comment:
I don't remember for sure, but this was probably done to avoid "uninitialized variable" warnings in optimized builds.

Reply:
I added a default constructor instead of relying on the users doing the zero-initialization.

Reply:
The default constructor doesn't zero-initialize, does it?

Reply:
It does because the member variables have `= 0;` in their declarations.

Reply:
I had missed that, sorry.